src/pages/permissions/actions/GroupActions.tsx

```diff
@@ -0,0 +1,114 @@
+import { FC, useRef, useState } from "react";
+import { ContextualMenu, Icon } from "@canonical/react-components";
+import { LxdGroup } from "types/permissions";
+import usePanelParams from "util/usePanelParams";
+import DeleteGroupModal from "./DeleteGroupModal";
+import usePortal from "react-useportal";
+import classnames from "classnames";
+
+interface Props {
+  group: LxdGroup;
+}
+
+const GroupActions: FC<Props> = ({ group }) => {
+  const panelParams = usePanelParams();
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+  const [displayAbove, setDisplayAbove] = useState(false);
+  const menuRef = useRef<HTMLImageElement>(null);
+
+  const adjustDisplayPosition = () => {
+    const menu = menuRef.current;
+    if (!menu) {
+      return;
+    }
+
+    const menuPosition = menu.getBoundingClientRect().top;
+    const showMenuAbove = menuPosition > window.innerHeight / 1.5;
+    setDisplayAbove(showMenuAbove);
+  };
```

Comment:
As discussed in the react guild, this should be done in react-components as a feature of the ContextualMenu component. But it can be a follow-up to improve later. Can you create a ticket for this to be improved?

Reply:
sure thing :+1: